### PR TITLE
Error page

### LINKED ISSUE #1194: Create an error page in Simorgh
**Is your feature request related to a problem? Please describe.**
Because Simorgh is a SPA it needs it an error page as the application can fail on the client side. The [error pages designs are on Zeplin](https://app.zeplin.io/project/5baa564c05c2eba0d2b89568?seid=5c5973040e57ce426c6f4331).

**Describe the solution you'd like**
We should have front-end components for the error pages. 
Our application should render 404 & 500 pages when these status codes are returned.

**Describe alternatives you've considered**
N/A

**Testing notes**
- Automation
Cypress tests to check the status code returns as `404`and the error printed within the "article" contains `404` & `Page cannot be found`.

- Manual
Manually test the article page to make sure there are no issues around overlapping and the colour displays as expected, especially across older devices/browsers.

- Accessibility
Test using AT to make sure the text on screen is being read out in the correct order. Before testing, confirm the correct order that the text should be read out in. We should also check the colour ratio levels are passing.

Dev insight: Unit tests would be needed, as should some simple integration test (just for the heading, not all the error page attributes). 

**Additional context**
On the server side, Mozart can handle some errors, see [issue](https://github.com/bbc/simorgh-infrastructure/issues/176)

- [x] Initially labelled with ["Refinement needed"](https://github.com/BBC-News/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by dr3**
Resolves #1194

**Overall change:** A MVP implementation of our error page design.

**Code changes:**

- Uses and adapts existing psammead components to build a simple implementation of error pages to be refactored later.
- Adds translations to service context to be refactored later
- Adds more snapshot tests for other status codes

image

image

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/containers/ErrorMain/index.jsx
Comment (ChrisBAshton) [inline src/app/containers/ErrorMain/index.jsx]: This has always bothered me 🤔feels so unnecessary for a static page.

I'd usually just map the array index to the id - it [is considered an antipattern sometimes](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318) but is safe to use in cases like ours:

screen shot 2019-02-18 at 09 31 46

Then again, the `nanoid` approach would at least be consistent across our codebase. So I'm 60:40 _for_ changing it to `index`. What do you think?

File: src/app/containers/ErrorMain/index.jsx
Comment (ChrisBAshton) [inline src/app/containers/ErrorMain/index.jsx]: Rather than using `aria-hidden`/`aria-label`, I'd structure the HTML like so:

```jsx
<h1>
  <span>{messaging.statusCode}</span>
  {messaging.title}
</h1>
```

Then your span can just have something like:

```css
h1 span {
  display: block;
  color: etc;
}
```

It looks like [JAWS may have a problem with your current implementation](http://davidmacd.com/blog/does-aria-label-override-static-text.html) (if I've understood that article right!)

File: src/app/containers/ErrorMain/index.jsx
Comment (ChrisBAshton) [inline src/app/containers/ErrorMain/index.jsx]: This works, but is kinda confusing for English (as there _is_ no `callToActionLast`) and doesn't scale if we need to add a second URL (we'd have `callToActionMiddleFirst` and `callToActionMiddleLast` and it can quickly get out of hand...!)

How we've approached the problem in VJ is like so:

```json
{
  "url": "https://example.com",
  "callToAction": "Why not [[visit the example site]]?"
}
```

(Or even as markdown: `Why not [visit the example site](https://example.com)?`)

Then in JS, something like:

```js
const linkText = callToAction.match(/\[\[(.)+\]\]/)[0];
return callToAction.replace(`[[${linkText}]]`, `<a href="${url}">${linkText}</a>`);
```

Admittedly this could be tricky in JSX 🤔 Feels a bit cleaner than manually gluing bits of HTML string together.

File: src/app/lib/config/services/news.js
Comment (ChrisBAshton) [inline src/app/lib/config/services/news.js]: Is this using `’` purely to avoid escaping the `'` in the JSON? 🤔

**Review by ChrisBAshton — CHANGES_REQUESTED** [review]
Good work 🎉 and very speedy!

My main feedback would be: shouldn't the presentational stuff live in an ErrorMain _component_ rather than _container_? The container should be minimal, just setting up the service context etc, so that we're just passing things like `statusCode` as props to the presentational component.

The rest of my comments were just thoughts as I was reading your code, would be interested to get your thoughts on them!

File: src/app/containers/ErrorMain/index.jsx
Comment (dr3) [reply src/app/containers/ErrorMain/index.jsx]: Hey, not sure how this would work when including the `-` character

File: src/app/containers/ErrorMain/index.jsx
Comment (dr3) [reply src/app/containers/ErrorMain/index.jsx]: This is an MVP to get it out the door, 
`* Adds translations to service context to be refactored later`
This is a basic implementation before we add a translation layer. I dont think added regex at this change is in our best interest and we should wait to investigate this in the translations work

File: src/app/lib/config/services/news.js
Comment (dr3) [reply src/app/lib/config/services/news.js]: This was copied from UX, I will check which character they want

File: src/app/containers/ErrorMain/index.jsx
Comment (dr3) [reply src/app/containers/ErrorMain/index.jsx]: I think to be consistent with the rest of the codebase keeping nanoid is preferable

File: src/app/containers/ErrorMain/index.jsx
Comment (ChrisBAshton) [reply src/app/containers/ErrorMain/index.jsx]: Where has that requirement come from?

File: src/app/containers/ErrorMain/index.jsx
Comment (ChrisBAshton) [reply src/app/containers/ErrorMain/index.jsx]: Fair enough.

File: src/app/containers/ErrorMain/index.jsx
Comment (dr3) [reply src/app/containers/ErrorMain/index.jsx]: Have updated to suggestion as dash isnt needed

**Comment by dr3** [comment]
Blocked on https://github.com/bbc/psammead/issues/289

File: src/app/containers/ErrorMain/index.jsx
Comment (bcmn) [inline src/app/containers/ErrorMain/index.jsx]: This could use `styled-component` V4's [polymorphic `as` prop](https://medium.com/styled-components/announcing-styled-components-v4-better-faster-stronger-3fe1aba1a112#84b4) in the map, & might be neater.

**Review by bcmn — COMMENTED** [review]
The behaviour of this locally looks really good. 

> My main feedback would be: shouldn't the presentational stuff live in an ErrorMain _component_ rather than _container_? The container should be minimal, just setting up the service context etc, so that we're just passing things like `statusCode` as props to the presentational component.

I'm not sure you've addressed or responded to this? From reviewing the code, I think this is the only thought I'd have.

File: src/app/containers/ErrorMain/index.jsx
Comment (dr3) [reply src/app/containers/ErrorMain/index.jsx]: Done :)

**Comment by greenc05** [comment]
@dr3 I'm not sure the hover and focus CSS is correct, thought this was 2px?

**Comment by greenc05** [comment]
Also @dr3 if you see the screen reader UX doc, we shouldn't be using CSS for the list bullets, as this will read out as 'bullet' on some screen readers. It’s recommended to use a list style of none in the CSS e.g. we add our own ‘bullet’, and in addition add aria roles to the list of ‘list’ and ‘listitem’. Maybe this is in another ticket elsewhere? https://app.zeplin.io/project/5baa564c05c2eba0d2b89568/screen/5c5972c44a2d813589f1be3e

**Comment by dr3** [comment]
Hey @greenc05, the link underline styling is coming from our InlineLink component so we would need to make an issue to fix it everywhere, and this is a MVP of the error page to get something that looks a a little better. Im happy to make issues to improve and review a11y though :)

**Comment by greenc05** [comment]
Thanks @dr3 😄

File: src/app/lib/config/services/news.js
Comment (bcmn) [inline src/app/lib/config/services/news.js]:
```suggestion
        title: 'Page cannot be found',
```

This doesn't match the UX specification, & leaves the title reading rather oddly.

**Review by bcmn — COMMENTED** [review]
Nitpicky comment, otherwise I'm ready to approve 👍

**Comment by greenc05** [comment]
On that note @bcmn What is the page title @dr3 ?

**Comment by dr3** [comment]
Have added helmet for an MVP implementation of the metadata. This will be refactored in https://github.com/bbc/simorgh/issues/1350 and tested via E2E metadata tests.

**Review by bcmn — DISMISSED** [review]

**Review by ChrisBAshton — DISMISSED** [review]
Code looks good. Moving to 'Ready for Test'.

As per the issue (#1194), we'll need to add an integration test, so this PR will have more commits added before it can be merged. James please arrange a time with Drew to work on writing tests together.

Ideally we'd probably want to open a "Error page tests" PR _into this PR_, so that the tests can be easily reviewed separately. Then we can merge that PR, immediately followed by this one.

**Pull request merged**
